libraries/chain/db_management.cpp

```diff
@@ -164,6 +164,15 @@ void database::open(
 
       if( !find(global_property_id_type()) )
          init_genesis(genesis_loader());
+      else
+      {
+         _p_core_asset_obj = &get( asset_id_type() );
+         _p_core_dynamic_data_obj = &get( asset_dynamic_data_id_type() );
```

Comment:
There is no FC_ASSERT that ensures that _p_core_asset_obj->dynamic_asset_data_id == asset_dynamic_data_id_type(), better do a regular lookup on _p_core_asset_obj->dynamic_asset_data_id

Reply:
`asset_dynamic_data_id_type()` was used in `account_object::process_fees()`:
https://github.com/bitshares/bitshares-core/blob/68a0ffa1aa7ffd66940c10578c300e61ad6aa71f/libraries/chain/account_object.cpp#L77-L79

IMHO we need to add a FC_ASSERT.